### PR TITLE
Split into Windows/Linux config

### PR DESCRIPTION
**Pull request by andvra**
To make it possible to build on Linux. WIP

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      "description": "Should be inherited by some other presets",
+      "binaryDir": "${sourceDir}/out/build/${presetName}",
+      "installDir": "${sourceDir}/out/install/${presetName}",
+      "hidden": true
+    },
+    {
+      "name": "windows-base",
+      "hidden": true,
+      "inherits": [ "base-config" ],
+      "toolchainFile": "c:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake",
```
Comment (HWiman-ICONIC) [inline CMakePresets.json:16]: absolute path in committed file. As we discussed, not ideal. Is it really needed? The vcpkg toolchain is default in VS at least

**Review by HWiman-ICONIC — CHANGES_REQUESTED** [review]
I do not quite understand all changes. Let´s have a talk next week so you can explain (again...) what the problem was using vcpkg for building wxwidgets on linux. If the problem "only" is that wxWebView could not compile, then I think it would be easier (and better for the community) to fix the vcpkg build. That is the reason for these changes, right? Not the ambition to build everything at once?

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,68 +33,153 @@ if(WIN32)
...
+    message(STATUS "Flags: ${CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS}")
+    message(STATUS "Generator: ${CMAKE_GENERATOR}")
+    message(STATUS "Compiler: ${CMAKE_CXX_COMPILER}")
+    add_definitions("-fPIC -fno-jump-tables")
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+    add_compile_options(-fno-jump-tables)
+    add_subdirectory("wxWidgets")
+endif()
+
+set(wxWidgets_USE_UNICODE OFF)
```
[the remark this answers is not in the document]
Comment (andvra) [reply CMakeLists.txt:101]: Not sure why we do it, so I didn't want to remove this line either. Can of course look into it a bit further

File: CMakeLists.txt [hunk 3]
```diff
@@ -54,68 +33,153 @@ if(WIN32)
...
+        TARGETS IconicWebMap
+        EXPORT IconicWebMapTargets
+        RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+        ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
+        LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
+    )
+else()
+    # On Linux, we build wxWidgets from source, and thus also have to (?) build its targets
+    install(
+        TARGETS IconicWebMap wxcore wxbase wxwebview wxxml wxstc
```
[the remark this answers is not in the document]
Comment (andvra) [reply CMakeLists.txt:135]: They are linked automatically.

[the remark this answers is not in the document]
Comment (andvra) [reply CMakeLists.txt:135]: The idea is that wxWidgets will be built once, and from source, on Linux platforms. Then all our repos (iconic-api, iconic-studio and wxWebMap) will use wxWebMap from there, so the version they use will always be same among applications.

File: CMakeLists.txt [hunk 4]
```diff
@@ -124,41 +188,49 @@ install(
...
+# Copy export file to install directory
 install(
     FILES "${ICONIC_WEBMAP_EXPORTS_DIR}/webmap_exports.h"
     DESTINATION "${INSTALL_INCLUDE_DIR}"
 )
 
-# Copy DLL:s to target directory
-if(WIN32)
+# Copy DLL:s to install directory
+#if(WIN32)
```
[the remark this answers is not in the document]
Comment (andvra) [reply CMakeLists.txt:198]: yes

File: CMakePresets.json (hunk 1 above)
Comment (andvra) [reply CMakePresets.json]: Line with toolchain file is now removed. We should set the CMAKE_TOOLCHAIN_FILE as an environment variable on developer machines. If we use a preset file, and toolchainFile is left out, CMAKE_TOOLCHAIN_FILE will be empty. So, set the environment variable instead

File: CMakeLists.txt [hunk 5]
```diff
@@ -54,68 +33,153 @@ if(WIN32)
...
+#    if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+#        message(FATAL_ERROR "git submodule update --init --recursive failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
+#    endif()
+#endif()
+
+#if(NOT EXISTS "${PROJECT_SOURCE_DIR}/wxwidgets/CMakeLists.txt")
+#    message(FATAL_ERROR "The submodules were not downloaded! GIT_SUBMODULE was turned off or failed. Please update submodules and try again.")
+#endif()
+
+function(dump_cmake_variables)
```
[the remark this answers is not in the document]
Comment (andvra) [reply CMakeLists.txt:69]: Not used

**Pull request closed without merging**